test/Npgsql.Benchmarks/TypeHandlers/TypeHandlerBenchmarks.cs

```diff
@@ -66,37 +73,18 @@ public decimal Value
             }
         }
 
-        public static decimal[] Values => new decimal[]
-        {
-            0.0000000000000000000000000001M,
-            0.000000000000000000000001M,
-            0.00000000000000000001M,
-            0.0000000000000001M,
-            0.000000000001M,
-            0.00000001M,
-            0.0001M,
-            1M,
-            10000M,
-            100000000M,
-            1000000000000M,
-            10000000000000000M,
-            100000000000000000000M,
-            1000000000000000000000000M,
-            10000000000000000000000000000M,
-        };
-
         [Benchmark]
-        public void Read()
+        public T Read()
         {
             _readBuffer.ReadPosition = 0;
-            _handler.Read(_readBuffer, _elementSize);
+            return _handler.Read<T>(_readBuffer, _elementSize);
         }
 
         [Benchmark]
         public void Write()
         {
             _writeBuffer.WritePosition = 0;
-            _handler.Write(_value, _writeBuffer, null);
+            _handler.WriteWithLengthInternal(_value, _writeBuffer, null, null, false);
```

Comment:
Any reason for calling `WriteWithLengthInternal` instead of `Write`?

Reply:
The method is overriden in `NpgsqlSimpleTypeHandler` and it throws `NotSupportedException`.